File: evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {


        stack<int> n; 

        for (string& t : tokens){
     
            if (!(t == "*" || t == "+" || t == "-" || t == "/")){
                n.push(stoi(t));
                continue; 
            }

            // Not a digit
            char op = t[0];
            int val = n.top();
            n.pop();

            switch(op){
                case('+'):
                    n.top() += val; 
                    break;
                case('-'):
                    n.top() -= val; 
                    break;
                case('*'):
                    n.top() *= val;
                    break;
                case('/'):
                    n.top() /= val;
                    break;
            }
        }

        return n.top();
    }   
        
         

};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        operators = ['+', '-', '/', '*']

        for element in tokens:
            if element not in operators:
                stack.append(element)
            else:
                secondNum = int(stack.pop())
                firstNum = int(stack.pop())
                match element:
                    case "+":
                        val = firstNum + secondNum
                    case "-":
                        val = firstNum - secondNum
                    case "/":
                        val = firstNum / secondNum
                    case "*":
                        val = firstNum * secondNum
                stack.append(val)
        return int(stack[0])